cookbooks/fb_networkmanager/recipes/default.rb

```diff
@@ -0,0 +1,113 @@
+#
+# Cookbook:: fb_networkmanager
+# Recipe:: default
+#
+# Copyright (c) 2020-present, Vicarious, Inc.
+# Copyright (c) 2020-present, Facebook, Inc.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+unless node.ubuntu? || node.debian?
+  fail 'fb_networkmanager: Only Ubuntu or Debian are currently supported'
+end
+
+if node['fb_users']
+  vpnhome = '/var/lib/openvpn/chroot'
+  FB::Users.initialize_group(node, 'nm-openvpn')
+  node.default['fb_users']['users']['nm-openvpn'] = {
+    'gid' => 'nm-openvpn',
+    # If /var/lib/openvpn doesn't exist yet, we can't create it in time
+    # but we want to create the user, so for the first run, create it
+    # with /tmp
+    'home' => ::File.directory?(vpnhome) ? vpnhome : '/tmp',
+    'shell' => '/usr/sbin/nologin',
+    'action' => :add,
+  }
+end
+
+packages = %w{
+  network-manager
+  openvpn
+  network-manager-openvpn-gnome
+}
+
+package packages do
+  only_if do
+    node['fb_networkmanager']['enable'] &&
+      node['fb_networkmanager']['manage_packages']
+  end
+  action :upgrade
+end
```

Comment:
I'd still like to see the VPN parts split out into a separate cookbook, but for now setting `node.default['fb_networkmanager']['manage_packages']` to `false` is a reasonable enough workaround.

Reply:
It definitely can't go in a separate cookbook, this cookbook knows how to make VPN entries and APIs calling APIs doesn't work.

I can make `enable_openvpn_packages` if we want, or make it a separate recipe... but I think a config value seems sane. Then I can something like

```
package "network manager packages" do
  only_if do
    node['fb_networkmanager']['enable'] &&
      node['fb_networkmanager']['manage_packages']
  end
  package_names lazy {
    p = %w{network-manager}
    if node['fb_networkmanager']['enable_open_vpn']
      p += %w{openvpn network-manager-openvpn-gnome}
    end
    p
  }
end
```